packages/apps/graph/src/services/chainweb-node/non-fungible-account-details.ts

```diff
@@ -0,0 +1,80 @@
+import type { IClient } from '@kadena/client';
+import { Pact, createClient } from '@kadena/client';
+import type { ChainId } from '@kadena/types';
+import { dotenv } from '@utils/dotenv';
+import { networkConfig } from '../..';
+import type { Guard } from '../../graph/types/graphql-types';
+import { PactCommandError } from './utils';
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type NonFungibleChainAccountDetails = {
+  id: string;
+  account: string;
+  balance: number;
+  guard: {
+    keys: string[];
+    pred: Guard['predicate'];
+  };
+};
+
+function getClient(chainId: string, networkId: string): IClient {
+  return createClient(
+    `${dotenv.NETWORK_HOST}/chainweb/0.0/${networkId}/chain/${chainId}/pact`,
+  );
+}
+
+export async function getNonFungibleAccountDetails(
+  tokenId: string,
+  accountName: string,
+  chainId: string,
+): Promise<NonFungibleChainAccountDetails | null> {
+  let result;
+  const networkId = (await networkConfig).networkId;
```

Comment:
Regarding the `apiVersion`

```suggestion
  const { networkId, apiVersion } = await networkConfig;
```

Reply:
Feedback applied 👍